doc/psidk/src/consensus.md

```diff
@@ -57,24 +69,53 @@ BFT consensus has the following properties:
+
 ### View Change
 
 Each producer maintains a view change timer to detect failure in the leader. The view change timer is restarted whenever irreversibility advances. When the view change timer expires, a producer broadcasts a `view_change_message` to all other producers and increments the current term, moving to the next leader.
 
-- After initiating a view change, a producer will not restart the view change timer until it sees 2f+1 view change messages.
-- Consecutive view changes during which the chain does not make progress, cause the view change timeout to increase by a flat amount for each view change after the first.
+- After initiating a view change, a producer will not restart the view change timer until it sees 2f+1 block producers are in the same term or later.
+- Consecutive view changes during which the chain does not make progress, cause the view change timeout to increase.
 - A view change will trigger immediately if the leader of the current term or at least f+1 other producers are in a later term. This view change is independent of the view change timer and may skip terms.
 
+View change messages are broadcast network-wide. Every node retains the best view change from each active producer and broadcasts it to its peers whenever it receives a better one. When a node changes its view of active producers, it requests the current list of view change messages from all its peers.
+
+The ordering of view change messages from a single node is defined as follows:
+- A higher term is always better than a lower term
+- The ordering of view changes with the same term depends on whether the node that issued the view change is the leader.
+  - If the source is not the leader, a view change is better if it refers to an older block
+  - If the source is the leader, a view change is better if it refers to a newer block
+
+Note: Since the condition for activating a view requires the leader's view change to refer to a block which not older than other nodes' view changes, this ordering guarantees that as long as a valid set of view changes exist, all connected, honest nodes will converge to the same valid set.
+
+### Safety
+
+Let block X and block Y be two conflicting blocks which are both irreversible
+- Without loss of generality, let X be better than Y
+- Y has been committed by 2f+1 producers, because that is the requirement for it to be considered irreversible
+- let Z be the earliest block that is an ancestor of X (inclusive of X), conflicts with Y, is better than Y, and is prepared by 2f+1 producers. Such a block is guaranteed to exist because X itself must be prepared by 2f+1 producers in order to be committed by any honest producer.
+
+The existence of Z implies that at least f+1 producers have violated the protocol
```

Comment:
Doesn't the existence of X and Y already imply that at least f+1 producers have violated the protocol?

Reply:
It does. This list is the breakdown of exactly how they violated the protocol.